src/traces/carpet/index.js

```diff
@@ -0,0 +1,31 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Carpet = {};
+
+Carpet.attributes = require('./attributes');
+Carpet.supplyDefaults = require('./defaults');
+Carpet.plot = require('./plot');
+Carpet.calc = require('./calc');
+Carpet.animatable = true;
+Carpet.calcPriority = 1;
+Carpet.defaultPriority = 1;
+
+Carpet.moduleType = 'trace';
+Carpet.name = 'carpet';
+Carpet.basePlotModule = require('../../plots/cartesian');
+Carpet.categories = ['cartesian', 'carpet', 'carpetAxis', 'notLegendIsolatable'];
+Carpet.meta = {
+    description: [
```

Comment:
Oh, I thought that ended up on https://plot.ly/javascript/reference, but I guess not currently. 

It intends to communicate some general info about the trace object especially on how different attribute can interact with one another, the [`heatmap`](https://github.com/plotly/plotly.js/blob/00355fb142ea41a36235676b5180ac9e23c5c9a8/src/traces/heatmap/index.js#L26) meta description, is probably the best we got so far.